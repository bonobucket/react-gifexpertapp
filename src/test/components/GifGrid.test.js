import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el gifGrid ', () => {

    test('Debe hacer match con el snapshot ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={'Goku'} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'sdfdsf97fs7f',
            title: 'Goku',
            url: 'https://localhost/cualquiera.jpg'
        },
        {
            id: 'ABC',
            title: 'Goku',
            url: 'https://localhost/cualquiera.jpg'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={'Goku'} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })




})
