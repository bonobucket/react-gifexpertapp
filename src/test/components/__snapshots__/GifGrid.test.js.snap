// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el gifGrid  Debe hacer match con el snapshot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el gifGrid  Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="sdfdsf97fs7f"
      key="sdfdsf97fs7f"
      title="Goku"
      url="https://localhost/cualquiera.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Goku"
      url="https://localhost/cualquiera.jpg"
    />
  </div>
</Fragment>
`;
